File: cpp/001-010/Longest Substring Without Repeating Characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_map<char, int> mymap;
        int start = 0;
        int ans = 0;
        
        for (int i = 0; i < s.length(); i++) {
            char ch = s[i];
            auto it = mymap.find(ch);
            if (it != mymap.end()) {
                start = max(start, it->second + 1);
            }
            
            mymap[ch] = i;
            ans = max(ans, i - start + 1);
        }
        
        return ans;
    }
};

File: golang/001-010/Longest Substring Without Repeating Characters.go
func lengthOfLongestSubstring(s string) int {
    var i, maximum_len int = 0, 0
    mymap := make(map[byte]int)
    
    for j := 0; j < len(s); j++ {
        mymap[s[j]]++
        for mymap[s[j]] == 2 && i < j {
            mymap[s[i]]--
            i++
        }
        
        if maximum_len < j - i + 1 {
            maximum_len = j - i + 1
        }
    }
    
    return maximum_len
}
